samcli/lib/cookiecutter/interactive_flow_creator.py

```diff
@@ -0,0 +1,80 @@
+""" This module parses a json/yaml file that defines a flow of questions to fulfill the cookiecutter context"""
+from typing import Dict, Optional, Tuple
+from samcli.commands._utils.template import get_template_data
+from .interactive_flow import InteractiveFlow
+from .question import Question, QuestionKind
+
+
+class InteractiveFlowCreator:
+    @staticmethod
+    def create_flow(flow_definition_path: str):
+        """
+        This method parses the given json/yaml file to create an InteractiveFLow. It expects the file to define
+        a list of questions. It parses the questions and add it to the flow in the same order they are defined
+        in the file, i.e. the default next-question of a given question will be the next one defined in the file,
+        while also respecting the question-defined _next_question map if provided.
+
+        Parameters:
+        ----------
+        flow_definition_path: str
+            A path to a json/yaml file that defines the questions of the flow. the file is expected to be in the
+            following format:
+            {
+                "Questions": [
+                    {
+                      "key": "key of the corresponding cookiecutter config",
+                      "question": "the question to prompt to the user",
+                      "kind": "the kind of the question, for example: confirm",
+                      "isRequired": true/false,
+                      # optional branching logic to jump to a particular question based on the answer. if not given
+                      # will automatically go to next question
+                      "nextQuestion": {
+                        "True": "key of the question to jump to if the user answered 'Yes'",
+                        "False": "key of the question to jump to if the user answered 'Yes'",
+                      }
+                    },
+                    ...
+                ]
+            }
+
+        Returns: InteractiveFlow(questions={k1: q1, k2: q2, ...}, first_question_key="first question's key")
+        """
+        questions, first_question_key = InteractiveFlowCreator._load_questions(flow_definition_path)
+        return InteractiveFlow(questions=questions, first_question_key=first_question_key)
+
+    @staticmethod
+    def _load_questions(flow_definition_path: str) -> Tuple[Dict[str, Question], str]:
+        previous_question: Optional[Question] = None
+        first_question_key: str = ""
+        questions: Dict[str, Question] = {}
+        template_data = get_template_data(flow_definition_path)
```

Comment:
Although `get_template_data()` works here, it might be worth just to use its implementation `yaml_parse(fp.read())` here. Otherwise customers might see error message like this `Template file not found at ...` when anything goes wrong, which is misleading.

Reply:
how is this possible? the method raise "Template file not found" only if `if not pathlib.Path(template_file).exists():` it raises "Failed to parse template" if anything else went wrong. Anyway, if there is a misleading behavior in the method, we can fix it, instead of defining a new method

Reply:
if you take a look at the implementation of `get_template_data()`, it is actually just open the file and pass to `yaml_parse()`. The reason I am calling this out is because `get_template_data()`'s name does not match what it is doing here.

We don't need to define a new method, we just need to do `yaml_parse()`